jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -878,7 +882,11 @@ else if (previous._status == Status.CHANGED)
                     break;
             }
         }
-        //Call the BulkListeners
+
+        // Call the ChangeSetListeners
+        reportChangeSet(changes);
```

Comment:
I would report the changes before the `DiscreteListener`s.
Is there a reason the changes are notified in between what was notified before?

Reply:
It is a bit strange to separate out the bulk changes from the change set.
Perhaps we should move the bulk to be before the discrete as well?

Reply:
... actually the other way to do this, is to do a single iteration through all the listeners.   If it is a bulk or set listener, simply call it; if it is a discrete listener, only then iterate through all the changes.

This lets the order of the listeners be controlled by the order of the adds, so if somebody really cares (which I doubt) they will be able to control it.

I also think it will result in less iteration over listeners, so it will be more efficient